src/adagrams.js

```diff
@@ -1,8 +1,142 @@
+    addLetters(nueve, 9);
+    addLetters(['E'], 12);
+
+    const drawnHand = _.sample(letterPool, 10);
+
+    ////Attempting to make sure the count doesn't go over the total amount available
+    // const drawnHand = []
+    // const draw = () => {
+    //   for (let i = 0; i < 10; i++) {
+    //     letter = _.sample(letterPool);
+    //     drawnHand.push(letter)
+    //     letterPool = _.difference(letterPool, letter)
+    //   }
+    // }
+    return drawnHand;
   },
+
+  usesAvailableLetters(input, lettersInHand) {
+
+    const inHandHash = {}
+    lettersInHand.forEach((letter) => {
+      inHandHash[letter] === undefined ? inHandHash[letter] = 1 : inHandHash[letter]++;
+    });
+
+    const inputLetters = input.toUpperCase().split("");
+
+    if (inputLetters.length > 10) {return false};
+
+    if (inputLetters === _.uniq(inputLetters)) {
+      _.isEmpty(_.difference(lettersInHand, inputLetters))
+    };
+
+    let valid = true;
+    inputLetters.forEach(checkLetter => {
+      if (valid === false) {return false;} 
+      else if (inHandHash[checkLetter] < 1) {
+        valid = false;
+      } else if (inHandHash[checkLetter] >= 1) {
+        inHandHash[checkLetter] -= 1;
+      } else {valid = false;}
+    });
+    return valid;
+  },
+
+  scoreWord(word) {
+    let points = 0;
+
+    const wordPoints = word.toUpperCase().split("");
+    if (wordPoints.length >= 7) {
+      points += 8;
+    }
+
+    const scoreChart = {};
+    const addPointValues = function (letters, value) {
+      letters.forEach((letter) => {
+        scoreChart[letter] = value;
+      });
+    }
+
+    const uno = ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'];
+    const due = ['D', 'G'];
+    const tre = ['B', 'C', 'M', 'P'];
+    const quattro = ['F', 'H', 'V', 'W', 'Y'];
+    const cinque = ['K'];
+    const otto = ['J', 'X'];
+    const dieci = ['Q', 'Z'];
+    addPointValues(uno, 1);
+    addPointValues(due, 2);
+    addPointValues(tre, 3);
+    addPointValues(quattro, 4);
+    addPointValues(cinque, 5);
+    addPointValues(otto, 8);
+    addPointValues(dieci, 10);
+
+    wordPoints.forEach((letter) => {
+      points += scoreChart[letter];
+    })
+    
+    return points;
+  },
+
+  highestScoreFrom(words){
+    
+    const score = {};
+    let highestScore = {
+      word: '',
+      score: 0,
+    };
+
+    words.forEach((word) => {
+      let topScore = Adagrams.scoreWord(word);
+      let topWord = word;
+      if (topScore > highestScore.score) {
+        highestScore.score = topScore;
+        highestScore.word = topWord;
+      }
+      else if (topScore === highestScore.score) {
+        if (topWord.length === 10 && highestScore.word.length !== 10) {
+          highestScore.score = topScore;
+          highestScore.word = topWord;
+        } else if (topWord.length < highestScore.word.length && highestScore.word.length !== 10) {
+          highestScore.score = topScore;
+          highestScore.word = topWord;
+        }
```

Comment:
Nice, clean logic! Consider breaking this out into it's own helper function called `tieBreaker`.